location/views.py

```diff
@@ -66,3 +68,52 @@ def review(request, pk):
             )
             r.save()
     return HttpResponseRedirect(reverse("location", args=(pk,)))
+
+
+@login_required
+def apartment_upload(request):
+    # if this is a POST request we need to process the form data
+    if request.method == "POST":
+        # create a form instance and populate it with data from the request:
+        form = ApartmentUploadForm(request.POST, request.FILES)
+        # check whether it's valid:
+        if form.is_valid():
+            # process the data in form.cleaned_data as required
+            city = form.cleaned_data["city"]
+            state = form.cleaned_data["state"]
+            address = form.cleaned_data["address"]
+            zipcode = form.cleaned_data["zipcode"]
+            rent_price = form.cleaned_data["rent_price"]
+            image = form.cleaned_data["image"]
+            suite_num = form.cleaned_data["suite_num"]
+
+            # TODO use geocoding API to get latitude and longitude
+
+            # create or retrieve an existing location
+            loc = Location.objects.get_or_create(
+                city=city, state=state, address=address, zipcode=zipcode
+            )[
+                0
```

Comment:
this is a total nitpick, but having the `[0]` spread out between 3 lines is super weird.

Reply:
Yeah I know it's really weird. I tried changing the formatting but black formatting didn't like it.